spec/datadog/appsec/processor_spec.rb

```diff
@@ -0,0 +1,227 @@
+# typed: ignore
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/processor'
+
+RSpec.describe Datadog::AppSec::Processor do
+  before do
+    skip 'disabled for Java' if RUBY_PLATFORM.include?('java')
```

Comment:
This spec currently hinges on libddwaf being available. I felt it required excessive mocking for little reason when it its not (which is currently only JRuby).